build.xml

```diff
@@ -1,126 +1,417 @@
+            <classpath refid="project.classpath" />
+            <compilerarg line="-s ${src_gen.dir}" />
+            <compilerarg line="-XDignore.symbol.file" />
+            <compilerarg line="-Xlint:all" />
+            <compilerarg line="--release 8" if:true="${is.atLeastJava9}" />
+        </javac>
+        <javac includeantruntime="false" srcdir="${src_gen.dir}" destdir="${classes.dir}" debug="true">
+            <classpath refid="project.classpath" />
+            <compilerarg line="-s ${src_gen.dir}" />
+            <compilerarg line="-Xlint:all" />
+            <compilerarg line="--release 8" if:true="${is.atLeastJava9}" />
+        </javac>
+        <javac includeantruntime="false" srcdir="${unit.dir}" destdir="${classes.dir}" debug="true">
+            <classpath refid="project.classpath" />
+            <compilerarg line="--release 8" if:true="${is.atLeastJava9}" />
+        </javac>
+
+        <travis target="compile" />
+    </target>
+
+    <target name="compile" depends="libs,compile-graalSOM" description="Compile GraalSOM and dependencies">
+    </target>
+
+    <target name="jar" depends="compile" description="Package as JAR">
+        <jar destfile="${build.dir}/graal-som.jar" basedir="${classes.dir}"></jar>
     </target>
 
     <target name="unit-tests" depends="compile" description="Execute tests">
         <junit haltonerror="false" haltonfailure="false" failureproperty="test.failed"
-            outputtoformatters="true">
+               outputtoformatters="true">
             <jvmarg value="-ea" />
             <jvmarg value="-esa" />
             <classpath refid="project.classpath" />
             <batchtest fork="yes" filtertrace="false">
-              <fileset dir="tests">
-                  <include name="**/*Test*.java"/>
-                  <include name="**/*TCK.java"/>
-                  <exclude name="dym"/>
-              </fileset>
+            <fileset dir="tests">
+                    <include name="**/*Test*.java"/>
+                </fileset>
             </batchtest>
             <formatter type="brief" usefile="false"/>
         </junit>
 
-        <java classname="som.vm.Universe" fork="true" failonerror="true">
+        <java classname="som.Launcher" fork="true" failonerror="true">
             <classpath refid="project.classpath" />
             <jvmarg value="-ea" />
             <jvmarg value="-esa" />
-            <arg line="-cp Smalltalk TestSuite/TestHarness.som" />
+            <arg line="-cp Smalltalk TestSuite/TestHarness.som  --ignore-inefficacies" />
         </java>
         <fail message="Basic tests failed." if="test.failed" />
     </target>
 
-    <target name="test" depends="unit-tests">
+    <target name="som-test" depends="compile" description="Test som script">
+        <travis target="som-test" start="SOM Test" />
+
+        <exec executable="./som" failonerror="true">
+            <arg value="TestSuite/TestHarness.som" />
```

Comment:
Here you don't seem to need the `--ignore-inefficacies` option.
There are two classes of assertions in unit tests: things that need to work, and things that should work, but not all implementations can provide it reliably.
Inefficacies are what I chose to call the cases that are not strictly required to pass.